app/controllers/admin/tags_controller.rb

```diff
@@ -9,7 +9,14 @@ def index
   end
 
   def create
-    Tag.category.create!(tag_params)
+    @existing_tag = Tag.where(name: tag_params.first).first
```

Comment:
The method `tag_params.first` is raising a warning:

```
DEPRECATION WARNING: Method first is deprecated and will be removed in Rails 5.1, as `ActionController::Parameters` no longer inherits from hash. Using this deprecated behavior exposes potential security problems. If you continue to use this method you may be creating a security vulnerability in your app that can be exploited. Instead, consider using one of these documented methods which are not deprecated: http://api.rubyonrails.org/v5.0.7.2/classes/ActionController/Parameters.html (called from create at /home/cjt/dev/consul/app/controllers/admin/tags_controller.rb:12)
```

How about `tag_params["name"]`? :thinking:

Maybe we could simplity the code and write:

```ruby
    Tag.find_or_create_by!(name: tag_params["name"]).update!(kind: "category")
```

What do you think?